docs/hello_nextflow/03_hello_containers.md

```diff
@@ -1,82 +1,442 @@
 
 ```bash
 exit
 ```
 
+Your prompt should now be back to what it was before you started the container.
+
 ### Takeaway
 
 You know how to pull a container and run it interactively, and you know how to use that to try out commands without having to install any software on your system.
 
 ### What's next?
 
-Learn how to use containers from within a workflow.
+Learn how to make data from your host system available to a container.
+
+## 2. Mounting data into containers
+
+When you run a container, it is isolated from the host system by default.
+This means that the container can't access any files on the host system unless you explicitly tell it to.
+One way to do this is to **mount** a **volume** from the host system into the container.
+
+Prior to working on the next section, confirm that you are in the `hello-nextflow` directory.
+
+```bash
+cd /workspace/gitpod/nf-training/hello-nextflow
+```
+
+### 2.1. Launch the container interactively with a mounted volume
+
+```bash
+docker run --rm -it --entrypoint /bin/sh -v $(pwd)/data:/data rancher/cowsay
+```
+
+This command mounts the `data` directory in the current working directory on the host system into the `/data` directory inside the container.
+We can explore the contents of the `/data` directory inside the container:
+
+```console title="Output"
+/ # ls
+bin    data   dev    etc    home   lib    media  mnt    opt    proc   root   run    sbin   srv    sys    tmp    usr    var
+/ # ls data
+bam              greetings.csv    ref              sample_bams.txt  samplesheet.csv
+```
+
+### 2.2. Use the mounted data
+
+Now that we have mounted the `data` directory into the container, we can use the `cowsay` command to display the contents of the `greetings.csv` file.
+
+```bash
+cat data/greetings.csv | cowsay
+```
+
+Output:
+
+```console title="Output"
+ _____________________
+< Hello,Bonjour,Holà >
+ ---------------------
+        \   ^__^
+         \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
+/ # exit
+```
+
+### Takeaway
+
+You know how to pull a container and run it interactively, and you know how to use that to try out commands without having to install any software on your system.
+
+### What's next?
+
+Learn how to get a container image for any pip/conda-installable tool.
 
 ---
 
-## 2. Use the container in a workflow
+## 2. Get a container image for a pip/conda-installable tool
+
+Some software developers provide container images for their software that are available on container registries like Docker Hub, but many do not.
+New Docker images can be created by writing a `Dockerfile` that specifies how to build the image, but this can be a lot of work.
+Instead, we can use the Seqera Containers web service to create a container image for us.
+
+### 2.1. Navigate to Seqera Containers
+
+Navigate to [Seqera Containers web service](https://www.seqera.io/containers/) and search for the `quote` pip package.
```

Comment:
Needs to tell us why we're getting this package. Generally speaking, briefly explain the use case / goal before you dive into the specifics of how/what to do.